DigitalLearningSolutions.Data/DataServices/PasswordDataService.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace DigitalLearningSolutions.Data.DataServices
+{
+    using System.Data;
+    using Dapper;
+
+    public interface IPasswordDataService
+    {
+        void SetPasswordByCandidateNumber(string candidateNumber, string passwordHash);
+        void SetPasswordByCandidateId(int candidateId, string passwordHash);
+    }
+
+    public class PasswordDataService: IPasswordDataService
+    {
+        private readonly IDbConnection connection;
+
+        public PasswordDataService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public void SetPasswordByCandidateNumber(string candidateNumber, string passwordHash)
+        {
+            connection.Query(
+                @"UPDATE Candidates
+                        SET Password = @passwordHash
+                        WHERE CandidateNumber = @candidateNumber",
+                new { passwordHash, candidateNumber });
+        }
+
+        public void SetPasswordByCandidateId(int candidateId, string passwordHash)

```

Comment:
Do we need both of these methods?

Reply:
I was making an assumption that when it comes to updating passwords elsewhere CandidateId would be a more useful identifier but that was probably too forward thinking. For this we only need the CandidateNumber one.